fabric/bin/peer.sh

```diff
@@ -332,39 +332,58 @@ handle_channel_join() {
 
     # - handle ercc
     say "Installing ercc on channel '${CHAN_ID}' ..."
-    echo "Packaging chaincode ..."
-    try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode package ercc.tar.gz -p ../../ercc --lang golang --label erccv1 
+    say "Packaging chaincode ..."
+    try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode package ${FABRIC_STATE_DIR}/ercc.tar.gz -p ../../ercc --lang golang --label erccv1 
     para
     sleep 3
-    echo "In installing chaincode ..."
-    try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode install ercc.tar.gz 
+    say "In installing chaincode ..."
+    try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode install ${FABRIC_STATE_DIR}/ercc.tar.gz 
     para
     sleep 3
-    echo "Query installed chaincodes.."
+    say "Query installed chaincodes.."
     try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode queryinstalled >&log.txt
     para
-    export PACKAGE_ID=$(awk '/Package ID:/{print}' log.txt | sed -n 's/^Package ID: //; s/, Label:.*$//;p')
-    echo "Approve for my org"
+    PACKAGE_ID=$(awk '/Package ID:/{print}' log.txt | sed -n 's/^Package ID: //; s/, Label:.*$//;p')
+    say "Approve for my org"
     try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode approveformyorg -o ${ORDERER_ADDR} --channelID ${CHAN_ID} --name ercc --version 1 --package-id ${PACKAGE_ID} --sequence 1 
     para
     sleep 3
-    echo "Check for commit readiness"
+    say "Check for commit readiness"
     try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode checkcommitreadiness --channelID ${CHAN_ID} --name ercc --version 1 --sequence 1 --output json 
     para
     sleep 3
-    echo "Commit chaincode definition...."
+    say "Commit chaincode definition...."
     try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode commit -o ${ORDERER_ADDR} --channelID ${CHAN_ID} --name ercc --version 1 --sequence 1 
     para
     sleep 3
-    echo "Query commited chaincodes on the channel"
+    say "Query commited chaincodes on the channel"
     try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode querycommitted --channelID ${CHAN_ID} 
-    #try $RUN ${FABRIC_BIN_DIR}/peer lifecycle chaincode querycommitted --channelID ${CHAN_ID} --name ercc
     para
     sleep 3
-    echo "call chaincode invoke..."
+    say "call chaincode invoke..."
     try $RUN ${FABRIC_BIN_DIR}/peer chaincode invoke -n ercc -c '{"function":"getSPID","args":[]}' -C ${CHAN_ID} 
     sleep 3
 
+    # TODO: Enable once tlcc is integrated
+    ##   - install ercc
+    #try $RUN ${FABRIC_BIN_DIR}/peer chaincode install -n ${ERCC_ID} -v ${ERCC_VERSION} -p github.com/hyperledger-labs/fabric-private-chaincode/ercc/cmd
+    #sleep 1
+    ##   - instantiate ercc iff "channel creation" peer
+    #if [ -e "${FABRIC_STATE_DIR}/${CHAN_ID}.creator" ]; then
```

Comment:
Hmm, what i meant in my prior comment related to channel-creator was not that copy as-is and comment out but you would have to do the put the `commit` in the same conditional statement as the `instantiate` in v1.4. Remember we want to replicate the same semantics we had before for v1.4 with v2 and `commit` essentially replaced `instantiate` ...

Reply:
I never really needed it so that's why I left in the comments.

Reply:
hmm, but you changed the semantics. We might not necessarily need it but the current commented out code as-is does not make sense (why would old lifecycle command show up in middle of new one) and also break the logic of creator. That's why i mentioned in my earlier code we either retain the original semantics -- where the code tried to do the instantiate (meaning commit now for new lifecycle) exactly once if you would have multiple peers -- or then just remove the creator logic throughout the code.